### PR TITLE
ENQT-294 Fix bug Category management

### PR DESCRIPTION
**Pull request by eugeniosant**

### CODE REVIEW

**Comment by entando-service** [comment]
All is well

File: admin-console/src/main/java/com/agiletec/apsadmin/category/CategoryAction.java [hunk 1]
```diff
@@ -260,31 +271,29 @@ protected void extractReferencingObjects(String categoryCode) {
...
+			if (this.getParentCategoryCode().contains(",") && !StringUtils.endsWith(this.getParentCategoryCode(), ",")) {
 				parentCategoryCode = StringUtils.trim(this.getParentCategoryCode().split(",")[1]);
 			} else {
 				parentCategoryCode = this.getParentCategoryCode();
 			}
 			if (this.getStrutsAction() == ApsAdminSystemConstants.EDIT) {
 				Category category = this.getCategory(this.getCategoryCode());
 				category.setTitles(this.getTitles());
-				category.setParentCode(parentCategoryCode);
+				//category.setParentCode(parentCategoryCode);
```
Comment (nicpuddu) [inline admin-console/src/main/java/com/agiletec/apsadmin/category/CategoryAction.java]: is this intentional?

File: admin-console/src/main/java/com/agiletec/apsadmin/category/CategoryAction.java [hunk 2]
```diff
@@ -260,31 +271,29 @@ protected void extractReferencingObjects(String categoryCode) {
...
+
 	public String save() {
 		String parentCategoryCode;
 		try {
-			if(this.getParentCategoryCode().contains(",") && !StringUtils.endsWith(this.getParentCategoryCode(), ",")) {
+			if (this.getParentCategoryCode().contains(",") && !StringUtils.endsWith(this.getParentCategoryCode(), ",")) {
 				parentCategoryCode = StringUtils.trim(this.getParentCategoryCode().split(",")[1]);
 			} else {
 				parentCategoryCode = this.getParentCategoryCode();
 			}
```
Comment (nicpuddu) [inline admin-console/src/main/java/com/agiletec/apsadmin/category/CategoryAction.java]: why not a ternary?
which method here can actually throw on exception? why inside the try - catch?

File: admin-console/src/main/java/com/agiletec/apsadmin/category/CategoryAction.java [hunk 3]
```diff
@@ -260,31 +271,29 @@ protected void extractReferencingObjects(String categoryCode) {
...
 			if (this.getStrutsAction() == ApsAdminSystemConstants.EDIT) {
 				Category category = this.getCategory(this.getCategoryCode());
 				category.setTitles(this.getTitles());
-				category.setParentCode(parentCategoryCode);
+				//category.setParentCode(parentCategoryCode);
 				this.getCategoryManager().updateCategory(category);
 				_logger.debug("Updated category {}", category.getCode());
+			} else if (!StringUtils.isEmpty(this.getParentCategoryCode())) {
```
Comment (nicpuddu) [inline admin-console/src/main/java/com/agiletec/apsadmin/category/CategoryAction.java]: we already defined a variable containing the parentCategoryCode, why do we have to go through the getter again?
btw why is the category a CSV?

File: admin-console/src/main/java/com/agiletec/apsadmin/category/category.xml [hunk 4]
```diff
@@ -25,6 +25,11 @@
             <result type="tiles">admin.Category.entry</result>
             <interceptor-ref name="entandoDefaultStack"><param name="requestAuth.requiredPermission">manageCategories</param></interceptor-ref>
         </action>
+		
+		<action name="openCloseCategoryTreeInEntry" class="categoryAction" method="buildTree">
+            <result type="tiles">admin.Category.entry</result>
```
Comment (nicpuddu) [inline admin-console/src/main/java/com/agiletec/apsadmin/category/category.xml]: fix indentation please

File: admin-console/src/main/java/com/agiletec/apsadmin/category/CategoryAction.java (hunk 2 above)
Comment (eugeniosant) [inline admin-console/src/main/java/com/agiletec/apsadmin/category/CategoryAction.java]: Fixed... useless code

File: admin-console/src/main/java/com/agiletec/apsadmin/category/CategoryAction.java (hunk 1 above)
Comment (eugeniosant) [inline admin-console/src/main/java/com/agiletec/apsadmin/category/CategoryAction.java]: Yes, but do not leave the line...

File: admin-console/src/main/java/com/agiletec/apsadmin/category/CategoryAction.java (hunk 3 above)
Comment (eugeniosant) [inline admin-console/src/main/java/com/agiletec/apsadmin/category/CategoryAction.java]: Fixed... removed useless code

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

**Pull request merged**